fastats/core/decorator.py

```diff
@@ -29,29 +28,48 @@ def fs(func):
 
     @wraps(func)
     def fs_wrapper(*args, **kwargs):
+
         debug = kwargs.get('debug')
         return_callable = kwargs.get('return_callable')
 
         with suppress(KeyError):
             del kwargs['return_callable']
 
+        # This deliberately mutates the kwargs.
+        # We don't want to have a fs-decorated function
+        # as a kwarg to another, so we undecorate it first.
+        for k, v in kwargs.items():
+            if hasattr(v, 'undecorated'):
+                kwargs[k] = v.undecorated
```

Comment:
Would be nice to have helper funcs for marking/unmarking this

Reply:
My feeling is that splitting this code into separate functions would lose clarity. This only applies to the decorator it's defined in, so I think this is fine, unless there's a valid use case for the helper functions.

Reply:
OK - let's refactor if there's more metadata we need to store